### PR TITLE
prepare release 2.14.0

### PR DESCRIPTION
**Pull request by joelsmith**
### Checklist

- [X] Commits are signed with Developer Certificate of Origin (DCO)

### CODE REVIEW

**Comment by joelsmith** [comment]
Smoke tested on OpenShift and vanilla Kubernetes

File: .github/workflows/main-build.yml [hunk 1]
```diff
@@ -73,7 +73,7 @@ jobs:
       - name: Setup Go
         uses: actions/setup-go@v5
         with:
-          go-version: '1.22'
+          go-version: '1.21'
```
[the remark this answers is not in the document]
Comment (joelsmith) [reply .github/workflows/main-build.yml:76]: I'm not sure how 1.22 got there in the 2.13.1 update. The `relprep.sh` script should have made it match the operand, which is 1.21, but perhaps it also pulled in too-new versions of libraries which required 1.22
https://github.com/kedacore/keda/blob/release/v2.13/go.mod
Unfortunately we didn't catch it. Anyway, the script seemed to work better for 2.14 and set it to 1.21 to match the operand and that's why the downgrade.  I don't know that there would be any problems with using 1.22 at this point, but I guess my preference is to match the operand as much as possible for a given version (even though that means we have the above awkward downgrade).

[the remark this answers is not in the document]
Comment (jkyros) [reply .github/workflows/main-build.yml:76]: ok cool, just making sure, thanks :smile:

**Pull request merged**